Merge master into develop and update sdkconfig.defaults

luar123 opened the pull request:
I have been using the dev branch for a few days now and it works great so far. But I need fixes from master.
Also updated the sdkconfig.defaults files for esp32 and s3. I am always using this flow:

```
rm sdkconfig
SDKCONFIG_DEFAULTS="sdkconfig.defaults;sdkconfig_louder_esp_my" idf.py menuconfig
idf.py build
```
where `sdkconfig_louder_esp_my` contains only my personal (wifi) and device specific settings and can be replaced by other files for different hardware. If target is esp32s3 `sdkconfig.defaults.esp32s3` is automatically loaded after `sdkconfig.defaults`.

Will rebase once #140 is merged.
And once this one is merged I have another PR for making light sleep work.

CarlosDerSeher commented:
So you sync master with develop?

CarlosDerSeher commented:
Did you test different sample rates? I think there has bin some deadlock issue with http task.

Also did you see my comment in #140?

luar123 commented:
That is a good point. Tested only 48kHz so far. Will test 44.1kHz as well. Any other settings I should test?

For me no comments are shown in #140.

CarlosDerSeher commented:
No I can't think of anything else. But if it is stable for you with 48k and for me with 44.1k I believe it should be ok for rhe other sample rates too

luar123 commented:
I quickly tested 44.1kHz too and additionally chunk sizes 20, 26, 36ms and flac/opus/pcm with 48kHz. Had no issues, just switching from flac to opus gave a memory allocation error and required a reset of the esp (without PSRAM).

luar123 commented:
Updated everything:
* includes all commits from master branch
* fixed soft volume: a define was missing in `dsp_processor.c`
* fixed some default settings in kconfig files:
  * Fix USE_BIQUAD_ASM: was `true` (which doesn't work and results in `n`), set it to `y` for esp32 and `n` for all others
  * Fix SNAPSERVER_USE_MDNS: was `true`, set to `y`
  * Fix USE_SAMPLE_INSERTION: was `true`, set to `y`
  * Change SNAPCLIENT_USE_INTERNAL_ETHERNET from `y` to `n`. I think default should be disabled, because otherwise it would crash if no ethernet module is connected.

luar123 commented:
Thanks for merging.
You used squash merge, right? This will lead to merge conflicts when we will merge develop back into master at some point (because the original commits from master are replaced by the new squash commit). So in this case the normal add merge commit way would have been better. Let me know if you want to reset/revert and repeat the merge, otherwise I will proceed as is.

CarlosDerSeher commented:
Oh didn't know about that. Yes we could redo it if possible. Resolving merge conflicts is a tedious job

luar123 commented:
You could run `git reset HEAD~1` and force push to revert it. Hopefully no one has pulled it already. Or create a revert commit. Then I could create the PR again.

Or we simply merge master again, there were not too many conflicts to solve.

CarlosDerSeher commented:
@luar123 Alright I've reverted to preserve history. We can restart with this
